### PR TITLE
ROS 2 Port (reopened)

### PR DESCRIPTION
**Pull request by henningkayser**
This is a cleaned up version of #41 (fixed, squashed and rebased onto melodic-devel).

@clalancette Do you still want a version bump to 1.0.0 or would it suffice to add a `ros2` to the tag?

### CODE REVIEW

**Comment by clalancette** [comment]
Thanks for picking this up!  I'd like to separate the work on the `melodic-devel` branch from the actual port, so I've updated the `ros2` branch to have all of the commits from `melodic-devel`.  It seems like GitHub isn't actually picking that up for some reason, so I'm going to close and reopen this PR to try and do that.  I'll also do a regular review.

**Comment by clalancette** [comment]
This looks generally good.

About the version number, we should definitely bump to 1.0.0.  We can't just add a tag to the version since the backend tooling can't handle it, so let's bump that up to 1.0.0, rebase this, and then I think we can merge this in.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,16 @@
 Changelog for package urdfdom_py
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+1.0.0 (2020-01-06)
```
Comment (clalancette) [inline CHANGELOG.rst]: Don't make any changes to the CHANGELOG by hand; that will get done automatically when I do a release.

**Comment by clalancette** [comment]
Looks good to me!  I'm going to merge it.  I'd like to #52 onto melodic and this branch, so I'll hold off doing a release until we get that done.

**Comment by henningkayser** [comment]
> Looks good to me! I'm going to merge it. I'd like to #52 onto melodic and this branch, so I'll hold off doing a release until we get that done.

Thanks!

**Comment by clalancette** [comment]
At long last, I've now released this for ROS 2 Dashing and Eloquent.  Thanks for the patience here.

**Pull request merged**